contracts/RockPaperScissors.sol

```diff
@@ -0,0 +1,268 @@
+        require(playerMove < 3, INVALID_MOVE_MSG);
+
+        game.opponentMove = playerMove;
+        game.expiryDate =  block.timestamp + FORFEIT_WINDOW;
+        fundMove(stake);
+        emit OpponentPlays(msg.sender, gameToken, playerMove);
+    }
+
+
+    function fundMove(uint256 stake) internal {
+        uint256 senderBalance = balances[msg.sender];
+
+        if (msg.value < stake) {
+            require(balances[msg.sender] >= stake - msg.value, "Insufficient balance");
+            balances[msg.sender] = SafeMath.sub(senderBalance, stake - msg.value);
+            return;
+        }
+
+        balances[msg.sender] = senderBalance.add(msg.value - stake);
+        return;
+    }
+
+    /**
+    * Once the opponent has made their move, the creator of the game must resolve to determine the winner
+    * and reward the winner.
+    * If the creator does not resolve the game then the opponent can collect their winnings
+     * after the forfeit window has expired.
+    */
+    function resolveGame(bytes32 gameToken, bytes32 secret, uint8 creatorMove) external whenNotPaused {
+
+        Game storage game = games[gameToken];
+
+        address creator = game.creator;
+        require(msg.sender == creator, ONLY_CREATOR_CALL_MSG);
+        require(creatorMove < 3, INVALID_MOVE_MSG);
+
+        address opponent = game.opponent;
+        require(opponent != NULL_ADDRESS , "Still awaiting opponent");
+
+        //Validate move
+        bytes32 expectedMoveHash = hashPlayerMove(gameToken, secret, creatorMove);
+        require(game.moveHash == expectedMoveHash, "Move and secret do not match");
+
+        uint8 opponentMove = game.opponentMove;
+        uint256 stake = game.stake;
+
+        //Draw
+        if(opponentMove == creatorMove){
+            resetGame(game);
+            emit Draw(gameToken);
+            balances[creator] = balances[creator].add(stake);
+            balances[opponent] = balances[opponent].add(stake);
+            return;
+        }
+
+        //Determine winner
+        address winner;
+        address loser;
+        uint256 winnings = stake * 2;
+        uint256 fee = stake * FEE_PERCENTAGE/100;
+
+        // If the creator wins then is rewarded by not paying a fee on the winnings
+        if ((opponentMove + 1) % 3 == creatorMove) {
+            winner = creator;
+            loser = opponent;
+            winnings = winnings + fee;
+            balances[creator] = balances[creator].add(winnings);
+        // If the creator loses then rewarded by being able to charge the opponent with a player's fee
+        } else {
+            winner = opponent;
+            loser = creator;
+            balances[opponent] = balances[opponent].add(winnings - fee);
+            balances[creator] = balances[creator].add(fee); // Creator rewarded
+        }
+
+        resetGame(game);
+        emit WinnerRewarded(gameToken, winner, loser, winnings, creatorMove, opponentMove);
```

Comment:
Listening to the events will be a strange experience, where the balance invariant seems to be broken.
You should aim for all the events at inception to be enough to reconstruct the current contract state.